test/pipeline.test.ts

```diff
@@ -0,0 +1,133 @@
+    const key1 = v4();
+    const key2 = v4();
+    const value1 = v4();
+    const value2 = v4();
+    const dictionaryName = v4();
+
+    await client
+      .pipeline()
+      .hset(dictionaryName, key1, value1)
+      .hset(dictionaryName, key2, value2)
+      .set(key1, value1)
+      .set(key2, value2)
+      .exec();
+
+    const results = await client
+      .pipeline()
+      .hget(dictionaryName, key1)
+      .hget(dictionaryName, key2)
+      .get(key1)
+      .get(key2)
+      .exec();
+
+    expect(results).toEqual([
+      [null, value1],
+      [null, value2],
+      [null, value1],
+      [null, value2],
+    ]);
+  });
+  it('should be able to run with batch load commands', async () => {
+    const key1 = v4();
+    const key2 = v4();
+    const value1 = v4();
+    const value2 = v4();
+    const dictionaryName = v4();
+
+    await client
+      .pipeline([
+        ['hset', dictionaryName, key1, value1],
+        ['hset', dictionaryName, key2, value2],
+        ['set', key1, value1],
+        ['set', key2, value2],
+      ])
+      .exec();
+
+    const results = await client
+      .pipeline([
+        ['hget', dictionaryName, key1],
+        ['hget', dictionaryName, key2],
+        ['get', key1],
+        ['get', key2],
+      ])
+      .exec();
+
+    expect(results).toEqual([
+      [null, value1],
+      [null, value2],
+      [null, value1],
+      [null, value2],
+    ]);
+  });
+
+  it('you should be able to mix commands', async () => {
+    const key1 = v4();
+    const key2 = v4();
+    const key3 = v4();
+    const key4 = v4();
+    const value1 = v4();
+    const value2 = v4();
+    const value3 = v4();
+    const value4 = v4();
+    const dictionaryName = v4();
+
+    await client
+      .pipeline([
+        ['hset', dictionaryName, key1, value1],
+        ['hset', dictionaryName, key2, value2],
+        ['set', key3, value3],
+        ['set', key4, value4],
+      ])
+      .hset(dictionaryName, key1, value1)
+      .hset(dictionaryName, key2, value2)
+      .set(key3, value3)
+      .set(key4, value4)
+      .exec();
+
+    const results = await client
+      .pipeline([
+        ['hget', dictionaryName, key1],
+        ['hget', dictionaryName, key2],
+        ['get', key3],
+        ['get', key4],
+      ])
+      .hget(dictionaryName, key1)
+      .hget(dictionaryName, key2)
+      .get(key3)
+      .get(key4)
+      .exec();
+
+    expect(results).toEqual([
+      [null, value1],
+      [null, value2],
+      [null, value3],
+      [null, value4],
+      [null, value1],
+      [null, value2],
+      [null, value3],
+      [null, value4],
+    ]);
+  });
+  it('throws an error when using an un supported command', async () => {
+    try {
+      await client.pipeline([['UNSUPPORTED_CMD']]).exec();
+    } catch (err) {
+      if (process.env.MOMENTO_ENABLED === 'true') {
+        expect(err).toEqual(
+          new Error('Un-Supported Command Passed: UNSUPPORTED_CMD')
```

Comment:
👍

Reply:
would probably be good to include the name of the command in the error message?

Reply:
this is just a test. I am including it :) `UNSUPPORTED_CMD` was the test name of unsupported command in my unit test lol vs IORedis just returns
`          new TypeError("Cannot read properties of undefined (reading 'apply')")`

in this scenario so we actually handling better IMO